### PR TITLE
Misc fixes for zone picking

### PR DESCRIPTION
**Pull request by simahawk**
1. fix issue w/ pkg qty widget w/out unique key for pkgs
2. change validation for partial picking: a package is now required
* front: change the scan widget message
* back: send back to same screen and show "package not found" message
3. fix issue w/ qty handling on set line destination reload (qty was lost due to pt 2)

### CODE REVIEW

File: shopfloor/services/zone_picking.py [hunk 1]
```diff
@@ -647,15 +656,30 @@ def set_destination(
...
+                    location,
+                )
+                if response:
+                    return response
+                else:
+                    # TODO we should have a better way to determine this.
+                    # Right now is based on the fact that set destination
+                    # returns a response if we have to redirect somewhere else
+                    # because of issues.
+                    pkg_moved = True
```
Comment (sebalix) [inline shopfloor/services/zone_picking.py]: It is wrong here, `_set_destination_location` and `_set_destination_package` return also a response for the `zero_check` case (which is not an error, just the normal next step for the operator if zero check is enabled).
So yes we should find another way to know that the package has been moved, I propose:

- for `_set_destination_location`, check if `move_line.qty_done == quantity and move_line.location_dest_id == location`
- for `_set_destination_package`, check if `move_line.qty_done == quantity and move_line.result_package_id == package and move_line.shopfloor_user_id == self.env.uid`

**Review by sebalix — CHANGES_REQUESTED** [review]

File: shopfloor/services/zone_picking.py (hunk 1 above)
Comment (simahawk) [reply shopfloor/services/zone_picking.py]: makes sense, I'll check

Comment (simahawk) [reply shopfloor/services/zone_picking.py]: @sebalix fixed here https://github.com/camptocamp/wms/pull/42/commits/e27a9a527ebd38ba68549bdee62ec2ebd6961a36

**Review by sebalix — APPROVED** [review]
Not sure to understand the change, but I suppose it is what we want :p

**Comment by simahawk** [comment]
> Not sure to understand the change, but I suppose it is what we want :p

updated the description to explain what we wanted :wink:

File: shopfloor_mobile/static/wms/src/scenario/zone_picking.js [hunk 2]
```diff
@@ -133,19 +133,27 @@ const template_mobile = `
                 :message="{body: 'Full order picking, no more operation.'}"
                 />
             <picking-summary
+                v-if="stata.data.move_line"
```
Comment (guewen) [inline shopfloor_mobile/static/wms/src/scenario/zone_picking.js]:
```suggestion
                v-if="state.data.move_line"
```

File: shopfloor_mobile/static/wms/src/scenario/zone_picking.js [hunk 3]
```diff
@@ -133,19 +133,27 @@ const template_mobile = `
...
                 :record="state.data.move_line.picking"
                 :records="[state.data.move_line]"
                 :records_grouped="picking_summary_records_grouped([state.data.move_line])"
                 :list_options="picking_summary_move_line_list_options([state.data.move_line])"
                 :key="make_state_component_key(['picking-summary'])"
                 />
             <item-detail-card
+                v-if="stata.data.move_line"
```
Comment (guewen) [inline shopfloor_mobile/static/wms/src/scenario/zone_picking.js]:
```suggestion
                v-if="state.data.move_line"
```

File: shopfloor_mobile/static/wms/src/scenario/zone_picking.js [hunk 4]
```diff
@@ -133,19 +133,27 @@ const template_mobile = `
...
                 :key="make_state_component_key(['detail-move-line-dest-pack', state.data.move_line.id])"
                 :record="state.data.move_line"
                 :options="{main: true, key_title: 'package_dest.name'}"
                 :card_color="utils.colors.color_for('screen_step_todo')"
                 class="mt-2"
                 />
+            <div class="no-line-found" v-if="stata.data.move_line">
```
Comment (guewen) [inline shopfloor_mobile/static/wms/src/scenario/zone_picking.js]:
```suggestion
            <div class="no-line-found" v-if="state.data.move_line">
```

Comment (simahawk) [reply shopfloor_mobile/static/wms/src/scenario/zone_picking.js]: WTF? Seems I accidentally committed a typo after testing it for real :sweat:

**Comment by simahawk** [comment]
@guewen fixed and squashed. Thanks!

**Review by guewen — APPROVED** [review]

**Pull request merged**